jvm-runtime/ftl-runtime/common/deployment/src/main/java/xyz/block/ftl/deployment/ModuleBuilder.java

```diff
@@ -74,6 +76,7 @@ public class ModuleBuilder {
     public static final DotName OFFSET_DATE_TIME = DotName.createSimple(OffsetDateTime.class.getName());
     public static final DotName GENERATED_REF = DotName.createSimple(GeneratedRef.class);
     public static final DotName EXPORT = DotName.createSimple(Export.class);
+    private static final Pattern NAME_PATTERN = Pattern.compile("^[A-Za-z_][A-Za-z0-9_]*$");
```

Comment:
The naming convention isn't quite this simple, but this is a good improvement. ie. this will allow `DNSServer`, but the FTL name for that would be `DnsServer`. I think this is very uncommon in the JVM though?

Reply:
I copied this from validator.go: https://github.com/TBD54566975/ftl/blob/637f20ae9551397bde1743fb81f087af8fc9d168/backend/schema/validate.go#L247

Is there stricter validation somewhere? I had a look but I did not seem to be able to find it.